src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -304,8 +304,12 @@ private function dumpValue($value)
      */
     private function getServiceCall($id, Reference $reference = null)
     {
-        if (null !== $reference && ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE !== $reference->getInvalidBehavior()) {
-            return sprintf('@?%s', $id);
+        if (null !== $reference) {
```

Comment:
can't we do ?
``` 
if (null === $reference) {
    return sprintf('@%s', $id);
}

switch...
```

Reply:
that would force duplicating the `return sprintf('@%s', $id);` line.